build/ArtifactBuilder/SecurityHelpers.cs

```diff
@@ -0,0 +1,199 @@
+        SubjectFormUnknown = 0x800b0003,        // Trust provider does not support the form specified for the subject
+        SubjectNotTrusted = 0x800b0004,         // Subject failed the specified verification action
+        FileNotSigned = 0x800B0100,         // TRUST_E_NOSIGNATURE - File was not signed
+        SubjectExplicitlyDistrusted = 0x800B0111,   // Signer's certificate is in the Untrusted Publishers store
+        SignatureOrFileCorrupt = 0x80096010,    // TRUST_E_BAD_DIGEST - file was probably corrupt
+        SubjectCertExpired = 0x800B0101,        // CERT_E_EXPIRED - Signer's certificate was expired
+        SubjectCertificateRevoked = 0x800B010C,     // CERT_E_REVOKED Subject's certificate was revoked
+        UntrustedRoot = 0x800B0109          // CERT_E_UNTRUSTEDROOT - A certification chain processed correctly but terminated in a root certificate that is not trusted by the trust provider.
+    }
+
+    static class SecurityHelpers
+    {
+        private static readonly IntPtr INVALID_HANDLE_VALUE = new IntPtr(-1);
+        // GUID of the action to perform
+        private const string WINTRUST_ACTION_GENERIC_VERIFY_V2 = "{00AAC56B-CD44-11d0-8CC2-00C04FC295EE}";
+
+        [DllImport("wintrust.dll", ExactSpelling = true, SetLastError = false, CharSet = CharSet.Unicode)]
+        static extern WinVerifyTrustResult WinVerifyTrust(
+            [In] IntPtr hwnd,
+            [In][MarshalAs(UnmanagedType.LPStruct)] Guid pgActionID,
+            [In] WinTrustData pWVTData
+        );
+
+        public static bool VerifyEmbeddedSignature(string fileName)
+        {
+            WinTrustFileInfo winTrustFileInfo = null;
+            WinTrustData winTrustData = null;
+
+            try
+            {
+                // specify the WinVerifyTrust function/action that we want
+                var action = new Guid(WINTRUST_ACTION_GENERIC_VERIFY_V2);
+
+                // instantiate our WinTrustFileInfo and WinTrustData data structures
+                winTrustFileInfo = new WinTrustFileInfo(fileName);
+                winTrustData = new WinTrustData(winTrustFileInfo);
+
+                var result = WinVerifyTrust(INVALID_HANDLE_VALUE, action, winTrustData);
+                return (result == WinVerifyTrustResult.Success);
+            }
+            finally
+            {
+                // free the locally-held unmanaged memory in the data structures
+                winTrustFileInfo?.Dispose();
+                winTrustData?.Dispose();
+            }
+        }
+
+        //public static bool CheckFile(string filename)
+        //{
+        //    // check digital signature
+        //    var ret = WinTrust.VerifyEmbeddedSignature(filename);
+        //    if (!ret) return false;
+
+        //    // do some other checks - for example verify the subject
+        //    var cert = new X509Certificate2(filename);
+        //    return cert.Verify() && cert.Subject == "foo";
+        //}
```

Comment:
Should this be removed too?

Reply:
I left in as a reminder, in case we decide that we need to do more with the cert than just a simple verification.